class_18_web/homework/stocks.py

```diff
@@ -0,0 +1,114 @@
+import base64
+import json
+import os
+
+import requests
+from alpha_vantage.timeseries import TimeSeries
+
+TWEETS_TYPES = ("recent", "popular")
+
+# for proper working there should be following env variable set up
+# it's of concatenated API key and API secret
+# os.environ["TWITTER_API"] = f"{_key}:{_sec}"
+
+# it is not that secure as twitter's one
+ALPHAVANTAGE_API_KEY = "YG3GMROI3RPTIZ23"
+
+
+class StockCollector:
+    """Gets daily stock data from AlphaVantage API (via python driver).
+    Main usage is to call `collect` with provided ticker(s).
+    You will get back last close, max and min for last 100 days
+
+    """
+
+    name = "price_data"
+
+    def __init__(self):
+        self.key = ALPHAVANTAGE_API_KEY
+
+    def collect(self, ticker):
+        try:
+            data, _ = TimeSeries(key=self.key).get_daily(ticker)
+        except ValueError:
+            return {}
+        data = [float(val["4. close"]) for val in data.values()]
+        return {
+            "last_price": data[0],
+            "p100d_min": min(data),
+            "p100d_max": max(data),
+        }
+
+
+class TweetsCollector:
+    """Collects tweets for a given stock ticker where if was mentions
+    in hash tag.
+    Requires env var TWITTER_API set up. It has to be a string as following:
+    'app API key' + ':' + 'app API secret'
+    TODO: check if Bearer token expired and request a new one (check on exp policy)
+
+    """
+
+    name = "tweets"
+
+    def __init__(self):
+        self.session = requests.Session()
+        self.auth_url = "https://api.twitter.com/oauth2/token"
+        self.tweets_url = "https://api.twitter.com/1.1/search/tweets.json"
+        self._api_creds = self._get_twit_api_creds()
+        self.bearer = self._get_bearer()
+
+    def _get_twit_api_creds(self):
+        # not sure this is secure to store it this way
+        api_key_secret = os.getenv("TWITTER_API", None)
```

Comment:
Saving secret keys in the environment variable is ok. Also, for this aim, we can integrate with keys managers, for example, [Vault](https://www.vaultproject.io/)